x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java

```diff
@@ -750,7 +749,7 @@ public void invalidateActiveTokensForRealmAndUser(
                         logger.warn("No tokens to invalidate for realm [{}] and username [{}]", realmName, username);
                         listener.onResponse(TokensInvalidationResult.emptyResult(RestStatus.OK));
                     } else {
-                        invalidateAllTokens(tokenTuples.stream().map(t -> t.v1()).collect(Collectors.toList()), listener);
+                        invalidateAllTokens(tokenTuples, listener);
```

Comment:
This looks like a behavioural change...
We're now invalidating refresh tokens, but we used to only invalidate access tokens?

Reply:
No this method always invalidate all tokens, access and refresh. 
https://github.com/elastic/elasticsearch/blob/5d693b1f770aa7348a0253ff3258a49f477789ed/x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java#L780-L787

The above invalidation first targets `refresh_token`, then `access_token`.

The problem is that it assumes every access token has a refresh token, which is not true for access tokens created by `client_credentials`. So the change here is to fix the NPE in this case.